### ISSUE TITLE
implement in pure keras 3

### ISSUE THREAD

**Issue by cpmpercussion** [cpmpercussion/keras-mdn-layer#60]
at the moment, this layer uses Tensorflow probability to handle the mixture of diagonal-covariance normal model and produce the loss function.

Tensorflow-probability is a big dependency and causes issues with future compatibility that I'm already encountering.

creating a pure Keras implementation would be a better way forward. 

We would need careful testing to make sure that the maths is correct as in the new version, we would be responsible for correctness of the loss function and mixture definitions.
